interfaces/devdoc/file_util_linux.md

```diff
@@ -0,0 +1,69 @@
+# file_util_linux
+
+## Overview
+
+Linux implementation of the file operation functions.
+
+## Exposed API 
+
+```c
+// Defines removed for space constraints
+
+typedef void *HANDLE;
+typedef void *LPSECURITY_ATTRIBUTES;
+
+MOCKABLE_FUNCTION(, HANDLE, CreateFileA, const char*, lpFileName, unsigned long, dwDesiredAccess, unsigned long, dwShareMode, LPSECURITY_ATTRIBUTES, lpSecurityAttributes, unsigned long, dwCreationDisposition, unsigned long, dwFlagsAndAttributes, HANDLE, hTemplateFile);
+MOCKABLE_FUNCTION(, bool, CloseHandle, HANDLE, handle_input);
+```
+
+### CreateFileA
+
+```c
+MOCKABLE_FUNCTION(, HANDLE, CreateFileA, const char*, lpFileName, unsigned long, dwDesiredAccess, unsigned long, dwShareMode, LPSECURITY_ATTRIBUTES, lpSecurityAttributes, unsigned long, dwCreationDisposition, unsigned long, dwFlagsAndAttributes, HANDLE, hTemplateFile);
+```
+
+`CreateFileA` implements the creating file function in Linux. Uses the `open` function in C for Linux.
+
+**SRS_FILE_UTIL_LINUX_09_001: [** If the `full_file_name` input is either empty or `NULL`, `CreateFileA` shall return an `INVALID_HANDLE_VALUE`. **]**
```

Comment:
>INVALID_HANDLE_VALUE [](http://example.com/codeflow?start=122&length=20)

I see this value is defined in the .h file. Please include it in this file so we know which value you are referring to. Space is not that constrained :)

Reply:
It would be nice to know all values HANDLE can take.

Reply:
I included the define for the INVALID_HANDLE_VALUE. But, I can't define the other values HANDLE will take because there would be hundreds.

Reply:
Does this module return any of those hundreds of values? We should have the ones that the module does return listed here. If the only value that the module explicitly returns is `INVALID_HANDLE_VALUE` then we're good as is.

Reply:
INVALID_HANDLE_VALUE is the only value that is explicitly returned by this module.